src/fu-engine.c

```diff
@@ -1079,11 +1079,29 @@ static gboolean
 fu_engine_check_requirement_firmware (FuEngine *self, XbNode *req,
 				      FuDevice *device, GError **error)
 {
+	FuDevice *device_actual = device;
+	guint64 depth;
 	g_autoptr(GError) error_local = NULL;
 
+	/* look at the parent device */
+	depth = xb_node_get_attr_as_uint (req, "depth");
+	if (depth != G_MAXUINT64) {
+		for (guint i = 0; i < depth; i++) {
+			device_actual = fu_device_get_parent (device_actual);
+			if (device_actual == NULL) {
+				g_set_error (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_SUPPORTED,
+					     "No parent device for %s (%u/%" G_GUINT64_FORMAT ")",
```

Comment:
I personally think that name is more useful than device id in an error message.

Reply:
Sure, that works too. Fixed.